src/test/java/org/redcrosswarriors/controller/TestRequestBloodController.java

```diff
@@ -1,67 +1,72 @@
-//import org.springframework.security.test.context.support.WithMockUser;
-//import org.springframework.test.web.servlet.MockMvc;
-//
-//import static org.mockito.Mockito.when;
-//import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
-//import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
-//import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
-//import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
-//
-//@SpringBootTest
-//@AutoConfigureMockMvc
-//public class TestRequestBloodController {
-//
-//
-//    @Autowired
-//    private MockMvc mockMvc;
-//
-//    @MockBean
-//    RequestBloodControllerService service;
-//
-//    @WithMockUser(username="admin@wayne.edu",roles={"USER","ADMIN"})
-//    @Test
-//    public void testRequestBlood() throws Exception
-//    {
-//        RequestBloodInput input  = new RequestBloodInput();
-//
-//        input.setFirstName("John");
-//        input.setLastName("Doe");
-//        input.setEmail("johhn@wayne.edu");
-//        input.setPhoneNumber("0000000000");
-//        input.setBloodType("AB+");
-//        input.setHospitalName("Beaumont");
-//        input.setStreetName("1234 XYZ");
-//        input.setCityName("ABC");
-//        input.setStateName("MI");
-//        input.setZipCode("48084");
-//        input.setMessage("I need Blood");
-//
-//        when(service.requestBlood(input)).thenReturn(true);
-//        Gson gson = new Gson();
-//        String json = gson.toJson(input);
-//        this.mockMvc.perform(post("/requestblood").contentType(MediaType.APPLICATION_JSON)
-//                .content(json)
-//                .accept(MediaType.APPLICATION_JSON)
-//        ).andDo(print()).andExpect(status().isOk());
-//
-//    }
-//}
\ No newline at end of file
+@SpringBootTest
+@AutoConfigureMockMvc
+public class TestRequestBloodController {
+
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    RequestBloodControllerService service;
+
+    @WithMockUser(username="admin@wayne.edu",roles={"USER","ADMIN"})
+    @Test
+    public void testRequestBlood() throws Exception {
+        RequestBloodInput input  = new RequestBloodInput();
+
+        input.setFirstName("John");
+        input.setLastName("Doe");
+        input.setEmail("johhn@wayne.edu");
+        input.setPhoneNumber("0000000000");
+        input.setBloodType("AB+");
+        input.setHospitalName("Beaumont");
+        input.setStreetName("1234 XYZ");
+        input.setCityName("ABC");
+        input.setStateName("MI");
+        input.setZipCode("48084");
+        input.setMessage("I need Blood");
+
+        Map<String, Object> responseObject = new HashMap<String, Object>();
+        responseObject.put("message", "Successfully requested");
+        when(service.requestBlood(input, "admin@wayne.edu")).thenReturn(new ResponseEntity<Object>(responseObject, HttpStatus.OK));
```

Comment:
You mocked the service object which means that whenever any method is called from the service object in your actual code it's not actually calling that method. To fix your test you need to add an additional when(service.isBloodTypeValid(input)).thenReturn(true) . Otherwise when the method is called it's returning nothing because it's mocked. So your unit test is failing.